src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.InstallStateReader.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.Sdk.Localization;
+using static Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader;
+using System.Runtime.Serialization;
+using Microsoft.Deployment.DotNet.Releases;
+
+#if USE_SYSTEM_TEXT_JSON
+using System.Text.Json;
+#else
+using Newtonsoft.Json;
+using JsonTokenType = Newtonsoft.Json.JsonToken;
+#endif
+
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    public partial class SdkDirectoryWorkloadManifestProvider
+    {
+        class InstallState
+        {
+            public string? WorkloadSetVersion { get; set; }
+            public WorkloadSet? Manifests { get; set; }
+        }
+
+        static class InstallStateReader
+        {
+            public static InstallState ReadInstallState(string installStatePath)
```

Comment:
This looks very similar to GetWorkloadVersionFromGlobalJson. Can we combine the two methods?

Reply:
I've refactored this so most of the common code is a method called in `JsonReader` now.